python_ta/reporters/core.py

```diff
@@ -271,6 +271,7 @@ def on_close(self, stats, previous_stats):
 
         Close the reporter's output stream if should_close_out is True.
         """
+        self.out.flush()
```

Comment:
Okay, you did what I asked you to do (thanks!) but I realized that a better place for this flushing is at the end of `print_messages`/`display_messages` for each reporter, so that the full report is flushed every time the check is run. Because each reporter type has different behaviour, I think it'll be easiest to just add this line to each reporter individually rather than in this parent class.